model/jpa/src/main/java/org/keycloak/models/jpa/session/PersistentUserSessionEntity.java

```diff
@@ -47,14 +47,14 @@
                 " AND sess.realmId = :realmId AND sess.userId = :userId ORDER BY sess.userSessionId"),
         @NamedQuery(name="findUserSessionsByClientId", query="SELECT sess FROM PersistentUserSessionEntity sess INNER JOIN PersistentClientSessionEntity clientSess " +
                 " ON sess.userSessionId = clientSess.userSessionId AND clientSess.clientId = :clientId WHERE sess.offline = :offline " +
-                " AND sess.userSessionId = clientSess.userSessionId AND sess.realmId = :realmId ORDER BY sess.userSessionId"),
-        @NamedQuery(name="findUserSessionsCountsByClientId", query="SELECT clientSess.clientId, count(clientSess) " +
-                " FROM PersistentUserSessionEntity sess INNER JOIN PersistentClientSessionEntity clientSess " +
-                " ON sess.userSessionId = clientSess.userSessionId " +
-                // find all available offline user-session for all clients in a realm
-                " WHERE sess.offline = :offline " +
-                " AND sess.userSessionId = clientSess.userSessionId AND sess.realmId = :realmId " +
-                " GROUP BY clientSess.clientId")
+                " AND sess.realmId = :realmId ORDER BY sess.userSessionId"),
```

Comment:
Why was the `sess.userSessionId = clientSess.userSessionId` part removed?

Reply:
Isn't the equality of `sess.userSessionId = clientSess.userSessionId` secured by the `INNER JOIN PersistentClientSessionEntity clientSess ON sess.userSessionId = clientSess.userSessionId` part?